File: src/main/python/visualphysics/layout/Size.py
#!/usr/bin/env python3.3

# Copyright 2012 by Douglas Sweetser, sweetser@alum.mit.edu
# Licensed under the Apache License, Version 2.0.

import argparse

'''Class Size
Used for pixel sizes used in layouts.
Author: sweetser@alum.mit.edu'''
class Size:

    def __init__(self, s=0):
        self.s = s

    def simple_print(self):
        result = str(self.s)
        print(result)
        return result

    def pretty_print(self):
        result = "The size is: " + str(self.s) + "."
        print(result)
        return result

if __name__ == '__main__':
    args_parser = argparse.ArgumentParser(description='Prints a point given two bits of data')
    args_parser.add_argument('-s', '--sprint', action='store_true', default=False, help="Simple print")
    args_parser.add_argument('-p', '--pprint', action='store_true', default=False, help="Pretty print, more verbose")
    args_stuff = args_parser.parse_known_args()
    args = args_stuff[0]
    argv = args_stuff[1]
 
    if (not args.pprint):
        args.sprint = True

    while argv:
        s = argv.pop(0)
        foo_size = Size(s)
        if args.sprint:
            foo_size.simple_print()
        if args.pprint:
            foo_size.pretty_print()

File: src/main/java/org/visualphysics/layout/Size.java
/* Copyright 2012 by Douglas Sweetser, sweetser@alum.mit.edu
 * Licensed under the Apache License, Version 2.0
 */

package org.visualphysics.layout;

import java.util.ArrayList;
import java.util.List;
import com.beust.jcommander.JCommander;

/** Class Size
 *  Used for x and y values used in layouts.
 *  @author sweetser@alum.mit.edu
 */
public class Size 
{
    int s;

    public Size(int s) 
    {
        this.s = s;
    }

    public String simple_print() 
    {
        String result = Integer.toString(this.s);
        System.out.println(result);
        return result;
    }

    public String pretty_print() 
    {
        String result = "The size is: " + Integer.toString(this.s) + ".";
        System.out.println(result);
        return result;
    }
    
    public static void main(String[] args) 
    {
        SizeArgs params = new SizeArgs();
        JCommander cmd = new JCommander(params, args);

        if(params.help) 
        {
            System.out.println("usage: Size.java [-h] [-s] [-p]\n\nPrints a size.\n\noptional arguments:\n  -h, --help    show this help message and exit\n  -s, --sprint  Simple print\n  -p, --pprint  Pretty print, more verbose");
        }

        if(params.pprint) 
        {
        }
        else 
        {
            params.sprint = true;
        } 
   
        while (!params.argv.isEmpty()) 
        {
            int s = params.argv.remove(0);
            Size foo = new Size(s);
            if (params.sprint) 
            {
                foo.simple_print();
            }
            if (params.pprint) 
            {
                foo.pretty_print();
            }
        }        
    }
}
